File: local_examples/local_string_demo.py
# EXAMPLE: local_string_demo
# HIDE_START
import redis

r = redis.Redis(host="localhost", port=6379, db=0, decode_responses=True)
# HIDE_END

# STEP_START set_get
res = r.set("mykey", "Hello Redis!")
print(res)
# >>> True
res = r.get("mykey")
print(res)
# >>> Hello Redis!
# REMOVE_START
assert res == "Hello Redis!"
r.delete("mykey")
# REMOVE_END
# STEP_END

# STEP_START incr
res = r.set("counter", "10")
print(res)
# >>> True
res = r.incr("counter")
print(res)
# >>> 11
res = r.incr("counter", 5)
print(res)
# >>> 16
# REMOVE_START
assert res == 16
r.delete("counter")
# REMOVE_END
# STEP_END

File: local_examples/local_string_demo.js
// EXAMPLE: local_string_demo
// HIDE_START
const redis = require('redis');
const client = redis.createClient();

await client.connect();
// HIDE_END

// STEP_START set_get
let res = await client.set('mykey', 'Hello Redis!');
console.log(res);
// >>> OK
res = await client.get('mykey');
console.log(res);
// >>> Hello Redis!
// REMOVE_START
console.assert(res === 'Hello Redis!');
await client.del('mykey');
// REMOVE_END
// STEP_END

// STEP_START incr
res = await client.set('counter', '10');
console.log(res);
// >>> OK
res = await client.incr('counter');
console.log(res);
// >>> 11
res = await client.incrBy('counter', 5);
console.log(res);
// >>> 16
// REMOVE_START
console.assert(res === 16);
await client.del('counter');
// REMOVE_END
// STEP_END
